README.md

```diff
@@ -4,7 +4,12 @@
 [![Carthage compatible](https://img.shields.io/badge/Carthage-compatible-4BC51D.svg?style=flat)](https://github.com/Carthage/Carthage)
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 
-Needle is a dependency injection (DI) system for Swift. Unlike other DI frameworks, such as [Cleanse](https://github.com/square/Cleanse), [Swinject](https://github.com/Swinject/Swinject), Needle encourages **hierarchical DI structure and utilizes code generation to ensure compile-time safety**. This allows us to develop our apps and make changes with confidence. **If it compiles, it works.** In this aspect, Needle is more similar to [Android Dagger](https://google.github.io/dagger/).
+Needle is a dependency injection (DI) system for Swift. Unlike other DI frameworks, such as [Cleanse](https://github.com/square/Cleanse), [Swinject](https://github.com/Swinject/Swinject), Needle encourages **hierarchical DI structure and utilizes code generation to ensure compile-time safety**. This allows us to develop our apps and make code changes with confidence. **If it compiles, it works.** In this aspect, Needle is more similar to [Android Dagger](https://google.github.io/dagger/).
+
+Needle aims to achieve the following primary goals:
+1. Provide high reliability by ensuring dependency injection code is compile-time safe.
+2. Ensure code generation is highly performant even when used with extremely large, multi-million-line codebases.
```

Comment:
Another nit: "extremely large, multi-million" could be considered redundant, since you've immediately qualified it with an actual number. 

Maybe reevaluate what you want to convey here? Is a multi-million line code base the normal user, or the exception? Is the goal to convince potential users that Needle will scale as their company grows?

Reply:
I'll remove the "extremely large" portion.